wordcloud.py

```diff
@@ -105,6 +103,3 @@ def sort_by_value(dictionary):
     """Sort dictionary by value in descending order."""
     sorted_dictionary = dict(sorted(dictionary.items(), key=operator.itemgetter(1, 0), reverse=True))
     return sorted_dictionary
-
-
-main()
```

Comment:
It's ok to call a function here like "main" or "test" or something. Normally you would use that function to test or experiment with your code. 

Note, `import` actually runs the python file. So if I do `import wordcloud.py` it would actually run `main()`! That's not what we want so the normal (and common) way to prevent that from happening is to use the following code instead: 

```
if __name__ == '__main__':
    main()
```

In PyCharm if you start typing the word "main" and hit enter it generates that code for you.

Reply:
I probably don't understand this yet, sorry.  I'll try to google it and see if I can get it.

Reply:
Consider a program with the following 2 files:

thing.py:
```
print("thing")
```

import_thing.py:
```
import thing

print("done")
```


If we run thing.py we expect the output to be "thing" right? This is the output:

```
thing
```

If we run import_thing.py we expect the output to be "done", right? But that's incorrect. This is the actual output:

```
thing
done
```

Why does it print "thing"? What is happening is that `import thing` actually _runs_ the code inside thing.py! _(I'm not sure why Python behaves this way, but it might be because of backwards compatibility.)_

That's usually not what we want so the way to prevent that from happening is to use `if __name__ == '__main__':`. It means "if we are running this file". Let's change thing.py to:

thing.py:
```

if __name__ == '__main__':
    print("thing")
```

Now if we run thing.py the output is still:

```
thing
```

But if we run import_thing.py the output is now:

```
done
```

That's better. 😄 

So it's quite common for developers to want to write code to test their python file. For example, if thing had a function called do_thing() we might want to test it, like so:

thing.py:
```

def do_thing(a, b):
    return a + b

if __name__ == '__main__':
    result = do_thing(1, 2)
    print(f"Expected 3, got {result}")
```

If we run thing.py the output is:

```
Expected 3, got 3
``` 

Simply run the file to test it. But by doing it this way someone else can import thing.py and use the do_thing() function without any test code being executed.

We can improve it further though. Notice that the variable "result" is actually a global variable. We are strongly encouraged to avoid global variables for a number of reasons (e.g. you can have issues name conflicts) by using a function that does the testing we avoid global variables. 

thing.py:
```

def do_thing(a, b):
    return a + b

def test_do_thing():
    result = do_thing(1, 2)  # Local variable is ok!
    print(f"Expected 3, got {result}")

if __name__ == '__main__':
    test_do_thing()
```

This was a long reply, but hopefully helps understand what's going on!

Reply:
Great description - thank you.  As we discussed on Slack, I'll try to experiment with this so I can see it in action.